src/test/java/org/terasology/logic/health/HealEventTest.java

```diff
@@ -0,0 +1,132 @@
+        modules.add("Health");
+        return modules;
+    }
+
+    @Before
+    public void initialize() {
+        entityManager = getHostContext().get(EntityManager.class);
+        time = getHostContext().get(Time.class);
+    }
+
+    @Test
+    public void healTest() {
+        HealthComponent healthComponent = new HealthComponent();
+        healthComponent.currentHealth = 0;
+        healthComponent.maxHealth = 100;
+
+        final EntityRef player = entityManager.create();
+        player.addComponent(new PlayerCharacterComponent());
+        player.addComponent(healthComponent);
+
+        player.send(new DoHealEvent(10));
+        assertEquals(player.getComponent(HealthComponent.class).currentHealth, 10);
+
+        player.send(new DoHealEvent(999));
+        assertEquals(player.getComponent(HealthComponent.class).currentHealth, 100);
+    }
+
+    @Test
+    public void healEventSentTest() {
+        TestEventReceiver<BeforeHealEvent> receiver = new TestEventReceiver<>(getHostContext(), BeforeHealEvent.class);
+        List<BeforeHealEvent> list = receiver.getEvents();
+        assertTrue(list.isEmpty());
+
+        HealthComponent healthComponent = new HealthComponent();
+        healthComponent.currentHealth = 0;
+        healthComponent.maxHealth = 100;
+
+        final EntityRef player = entityManager.create();
+        player.addComponent(new PlayerCharacterComponent());
+        player.addComponent(healthComponent);
+
+        player.send(new DoHealEvent(10));
+        assertEquals(1, list.size());
+    }
+
+    @Test
+    public void healEventCancelTest() {
+        HealthComponent healthComponent = new HealthComponent();
+        healthComponent.currentHealth = 0;
+        healthComponent.maxHealth = 100;
+
+        final EntityRef player = entityManager.create();
+        player.addComponent(new PlayerCharacterComponent());
+        player.addComponent(healthComponent);
+
+        TestEventReceiver<BeforeHealEvent> receiver = new TestEventReceiver<>(getHostContext(), BeforeHealEvent.class,
+        (event, entity) -> {
+            event.consume();
+        });
+        player.send(new DoHealEvent(10));
+        assertEquals(0, player.getComponent(HealthComponent.class).currentHealth);
+    }
+
+    @Test
+    public void healModifyEventTest() {
+        HealthComponent healthComponent = new HealthComponent();
+        healthComponent.currentHealth = 0;
+        healthComponent.maxHealth = 100;
+
+        final EntityRef player = entityManager.create();
+        player.addComponent(new PlayerCharacterComponent());
+        player.addComponent(healthComponent);
+
+        TestEventReceiver<BeforeHealEvent> receiver = new TestEventReceiver<>(getHostContext(), BeforeHealEvent.class,
+                (event, entity) -> {
+                    event.add(10);
+                    event.subtract(5);
+                    event.multiply(2);
+                });
+        // expected = ( initial:10 + 10 - 5 ) * 2 = 30
```

Comment:
I'd rewrite this ever so slightly for clarity. Like:

```suggestion
        // Expected value is ( initial:10 + 10 - 5 ) * 2 == 30
```